platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/CryptoBench.java

```diff
@@ -201,8 +203,17 @@ public void transferPrefetch() throws Exception {
         final long[] map = new long[verify ? maxKey : 0];
         VirtualMap<BenchmarkKey, BenchmarkValue> virtualMap = createMap(map);
 
-        final ExecutorService prefetchPool =
-                Executors.newCachedThreadPool(new ThreadConfiguration(getStaticThreadManager())
+        // Use a custom queue and executor for warmups. It may happen that some warmup jobs
+        // aren't complete by the end of the round, so they will start piling up. To fix it,
+        // clear the queue in the end of each round
+        final BlockingQueue<Runnable> queue = new LinkedBlockingQueue<>();
```

Comment:
Not strictly necessary in this code, but it still might be good to set a fixed capacity on this queue, to ensure it doesn't grow overly large.  Even 16384 or 32768 would be sufficient to prevent unbounded queue issues.

Reply:
First, this is benchmark code, not a part of platform runtime. Second, every round the queue is cleared, and a finite number of jobs is submitted to the thread pool (numRecords * 2, where numRecords is a benchmark param, usually 10_000 or 100_000). I mean, even if the thread pool contains a single thread, queue size may not exceed a very well defined limit. Finally, this is a pool of prefetch jobs, which are all optional and may be skipped. At the same time, I don't want to block the main thread or handle rejected execution exceptions in the loop right below...